### PR TITLE
index option for adding primitive to PrimitiveCollection

### PR DESCRIPTION
**Pull request by hpinkos**
I have a use case where I need to add a primitive to the collection at index 0.  This mimics what `ImageryLayerCollection` does.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @hpinkos!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/primitive-index/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Scene/PrimitiveCollection.js [hunk 1]
```diff
@@ -100,18 +100,29 @@ define([
...
+
         //>>includeStart('debug', pragmas.debug);
         if (!defined(primitive)) {
             throw new DeveloperError('primitive is required.');
         }
+        if (hasIndex) {
+            if (index < 0) {
+                throw new DeveloperError('index must be greater than or equal to zero.');
+            } else if (index > this._primitives.length) {
+                throw new DeveloperError('index must be less than or equal to the number of layers.');
```
Comment (likangning93) [inline Source/Scene/PrimitiveCollection.js]: `number of primitives`?

**Comment by hpinkos** [comment]
Thanks @likangning93, ready!

**Comment by likangning93** [comment]
Does this also warrant an update to `CHANGES.md`?

**Comment by hpinkos** [comment]
Sure! why not

**Pull request merged**